src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,6 +596,32 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+
+        internal bool ReturnsAwaitableToVoidOrInt(CSharpCompilation compilation)
+        {
+            // Early bail so we only even check things that are System.Threading.Tasks.Task(<T>)
+            if (ReturnType is NamedTypeSymbol namedType &&
+                !(namedType.ConstructedFrom == compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_Task) ||
+                  namedType.ConstructedFrom == compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_Task_T)))
+            {
+                return false;
+            }
+
+
+            var dumbInstance = new BoundLiteral(CSharpSyntaxTree.Dummy.GetRoot(), ConstantValue.Null, ReturnType);
+            var syntax = this.ExtractReturnTypeSyntax();
+            var binder = compilation.GetBinder(syntax);
```

Comment:
>var binder = compilation.GetBinder(syntax); [](start = 12, length = 43)

Please add a prototype comment that we might need to adjust the containing member of the binder to be the Main method.